include/SDL_pen.h

```diff
@@ -0,0 +1,316 @@
+ *    whenever "which" == ::SDL_PEN_MOUSEID.
+ *
+ *  This header file describes advanced functionality that can be useful for managing user configuration
+ *    and understanding the capabilities of the attached pens.
+ *
+ *  We primarily identify pens by ::SDL_PenID.  The implementation makes a best effort to relate each :SDL_PenID
+ *    to the same physical device during a session.  Formerly valid ::SDL_PenID values remain valid
+ *    even if a device disappears.
+ *
+ *  For identifying pens across sessions, the API provides the type ::SDL_PenGUID .
+ */
+
+#ifndef SDL_pen_h_
+#define SDL_pen_h_
+
+#include "SDL_stdinc.h"
+#include "SDL_error.h"
+#include "SDL_joystick.h" /* For SDL_JoystickGUID */
+
+#include "begin_code.h"
+/* Set up for C function definitions, even when using C++ */
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef struct SDL_PenID {       /**< SDL_PenIDs identify pens uniquely within a session */
+    Uint32 id;
+} SDL_PenID;
+
+#define SDL_PENID_ID_INVALID ((Uint32)0)
+#define SDL_PENID_VALID(penid) ((penid).id != SDL_PENID_ID_INVALID) /**< Test if the ::SDL_PenID is valid */
+
+typedef SDL_JoystickGUID SDL_PenGUID;   /**< UUID for pens, suitable for pesisting across sessions */
+
+#define SDL_PEN_MOUSEID ((Uint32)-2)    /**< Device ID for mouse events triggered by pen events */
+
+#define SDL_PEN_INFO_UNKNOWN (-1)       /**< Marks unknown information when querying the pen */
+
+/**
+ * \defgroup SDL_PEN_AXES Pen axis indices
+ *
+ * Below are the valid indices to the "axis" array from ::SDL_PenMotionEvent and ::SDL_PenButtonEvent.
+ * The axis indices form a contiguous range of ints from 0 to ::SDL_PEN_AXIS_LAST, inclusive.
+ * All "axis[]" entries are normalised to either 0..1 (unidirectional axes) or between -1..1 (bidirectional axes).
+ * Unsupported entries are always "0.0f".
+ *
+ * @{
+ */
+#define SDL_PEN_AXIS_PRESSURE     0 /**< Pen pressure.  Unidirectional: 0..1.0 */
+#define SDL_PEN_AXIS_XTILT        1 /**< Pen horizontal tilt fraction.  Bidirectional: -1.0..1.0 (left-to-right).  This is NOT the angle, but the tilt vector x component.  The physical max/min tilt may be smaller than -1.0 / 1.0, cf. \link SDL_PenCapabilityInfo \endlink .  */
+#define SDL_PEN_AXIS_YTILT        2 /**< Pen vertical tilt fraction.  Bidirectional: -1.0..1.0 (top-to-bottom).  This is NOT the angle, but the tilt vector y component.  */
+#define SDL_PEN_AXIS_DISTANCE     3 /**< Pen distance to drawing surface.  Unidirectional: 0.0..1.0 */
+#define SDL_PEN_AXIS_ROTATION     4 /**< Pen barrel rotation.  Unidirectional: -0..1.0 (clockwise, 0 is facing up).  Multiply with "2.0 * M_PI" to get radians, or "360.0" for degrees. */
+#define SDL_PEN_AXIS_SLIDER       5 /**< Pen finger wheel or slider (e.g., Airbrush Pen).  Unidirectional: 0..1.0 */
+
+#define SDL_PEN_AXIS_LAST         SDL_PEN_AXIS_SLIDER   /**< Last valid axis index */
```

Comment:
Why isn't this an enumeration, like the game controller axes?

Reply:
No strong preference; `#define` gives slightly better checking in C (if redefined).  Will switch to `enum`.